crates/duckdb-loadable-macros/src/lib.rs

```diff
@@ -1,11 +1,84 @@
 #![allow(clippy::redundant_clone)]
-use proc_macro2::{Ident, Span};
+use proc_macro2::{Ident, Literal, Punct, Span};
 
 use syn::{parse_macro_input, spanned::Spanned, Item};
 
 use proc_macro::TokenStream;
 use quote::quote_spanned;
 
+use darling::{ast::NestedMeta, Error, FromMeta};
+use syn::ItemFn;
+
+/// For parsing the arguments to the duckdb_entrypoint_c_api macro
+#[derive(Debug, FromMeta)]
+struct CEntryPointMacroArgs {
+    #[darling(default)]
+    /// The name to be given to this extension. This name is used in the entrypoint function called by duckdb
+    ext_name: String,
+    /// The minimum C API version this extension requires. It is recommended to set this to the lowest possible version
+    /// at which your extension still compiles
+    min_duckdb_version: Option<String>,
+}
+
+/// Wraps an entrypoint function to expose an unsafe extern "C" function of the same name.
+/// Warning: experimental!
+#[proc_macro_attribute]
+pub fn duckdb_entrypoint_c_api(attr: TokenStream, item: TokenStream) -> TokenStream {
+    let attr_args = match NestedMeta::parse_meta_list(attr.into()) {
+        Ok(v) => v,
+        Err(e) => {
+            return TokenStream::from(Error::from(e).write_errors());
+        }
+    };
+
+    let args = match CEntryPointMacroArgs::from_list(&attr_args) {
+        Ok(v) => v,
+        Err(e) => {
+            return TokenStream::from(e.write_errors());
+        }
+    };
+
+    // Set the minimum duckdb version (dev by default)
+    let minimum_duckdb_version = match args.min_duckdb_version {
+        Some(i) => i,
+        None => "dev".to_string(),
+    };
+
+    let ast = parse_macro_input!(item as syn::Item);
+
+    match ast {
+        Item::Fn(mut func) => {
+            let c_entrypoint = Ident::new(format!("{}_init_c_api", args.ext_name).as_str(), Span::call_site());
+            let original_funcname = func.sig.ident.to_string();
+            let prefixed_original_function = func.sig.ident.clone();
+
+            quote_spanned! {func.span()=>
+
+                /// # Safety
+                ///
+                /// Will be called by duckdb
+                #[no_mangle]
+                pub unsafe extern "C" fn #c_entrypoint(info: ffi::duckdb_extension_info, access: ffi::duckdb_extension_access) {
+                    let res = ffi::duckdb_rs_extension_api_init(info, access, #minimum_duckdb_version);
+
+                    if let Err(x) = res {
+                        // Error will be handled by DuckDB
```

Comment:
it will?

Reply:
Extensions should not throw or exit the program on error, but instead use the method in the extension access struct to set an error, which duckdb will then throw as a C++ exception https://github.com/duckdb/duckdb/blob/2d0842be932e59a887c5decc79ba9d50842be7f7/src/main/extension/extension_load.cpp#L537

Reply:
But now that I think about it I think about it, this is indeed incorrect at the moment:

I should just remove the error potentially returned from the `duckdb_rs_extension_api_init` function:
```rust
if #p_api.is_null() {
    return Err("DuckDB passed a nullpointer while trying to initialize the extension");
}
```

This case does indeed require no extra handling: DuckDB will only return a nullpointer if there is some error. But other parts of the initialization rust side might fail, which requires setting the error using the `extension_access` struct which i'm not doing here yet